packaging/docker/alpine-linux-k8s/Dockerfile

```diff
@@ -0,0 +1,46 @@
+FROM alpine:3.16.1 AS base
+
+RUN apk update && apk add --no-cache \
+    bash \
+    curl \
+    docker-cli \
+    docker-compose \
+    git \
+    jq \
+    libc6-compat \
+    openssh-client \
+    perl \
+    py-pip \
+    rsync \
+    run-parts \
+    su-exec \
+    tini \
+    tzdata
+
+FROM base AS downloader
+
+WORKDIR /downloads
+
+RUN curl -fL "https://storage.googleapis.com/kubernetes-release/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl" -o kubectl && \
+    chmod +x kubectl
+
+RUN curl -s "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh" | bash -s "/downloads"
+
+FROM base AS runtime
+
+COPY --from=downloader /downloads/kubectl /usr/local/bin/kubectl
+COPY --from=downloader /downloads/kustomize /usr/local/bin/kustomize
+
+ENV BUILDKITE_AGENT_CONFIG=/buildkite/buildkite-agent.cfg
+
+RUN mkdir -p /buildkite/builds /buildkite/hooks /buildkite/plugins \
+    && curl -Lfs -o /usr/local/bin/ssh-env-config.sh https://raw.githubusercontent.com/buildkite/docker-ssh-env-config/master/ssh-env-config.sh \
+    && chmod +x /usr/local/bin/ssh-env-config.sh
+
+COPY ./buildkite-agent.cfg /buildkite/buildkite-agent.cfg
```

Comment:
Because of how we mount the ConfigMap volumes in kubernetes it'd be useful if the config file and environment variable were at:
`/buildkite/config/buildkite-agent.cfg`

The volume mounts basically expect to own everything in the directory it mounts to and because `/buildkite` already has subdirectories for `/buildkite/builds`, `/buildkite/hooks`, and `/buildkite/plugins` we override the config file location in the agent deployment configuration. We could continue to do it in the deployment configuration -- but it might make more sense to do it here in the stack-specific agent configuration.

Reply:
Good idea! But I reckon we can do that as a next step.